pkgs/development/tools/pnpm/fetch-deps/default.nix

```diff
@@ -32,8 +32,14 @@
             outputHash = "";
             outputHashAlgo = "sha256";
           };
-      installFlags = lib.optionalString (pnpmWorkspace != "") "--filter=${pnpmWorkspace}";
+
+      filterFlags = lib.map (package: "--filter=${package}") pnpmWorkspaces;
     in
+    # pnpmWorkspace was deprecated, so throw if it's used.
+    assert (lib.throwIf (args ? pnpmWorkspace)
```

Comment:
Nit: I think it would look and feel better if you'd use something like:

```
assert assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
```

Reply:
Can you give a more specific example? We've tried `args.pnpmWorkspace != null` but it caused this issue: https://github.com/NixOS/nixpkgs/pull/350751#discussion_r1813860174

Reply:
FWIW even with `assertMsg` you need a `true` on the end. I’m not sure what the difference between `throwIf` and `assertMsg` is beyond a negation.